src/game/ui.py

```diff
@@ -22,24 +22,41 @@ def prompt_piece_selection(self, game_state):
                 game_state.message = "You must choose a number available in the list"
             except ValueError:
                 game_state.message = "You have to type number between 1 and " + str(PIECES_NUMBER)
-                self.display_game(game_state)
+                self.display_game(game_state, players)
```

Comment:
It's strange to not integrate the players into the game_state, why have you choice this architecture ?

Reply:
State represent the state of the current round game.
Players remains between rounds

Reply:
You can still integrate player into the game state even if they don't change. This way, you manipulate one state only and it's easier to reason about and test